webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -547,10 +547,10 @@ const AdvancedTextEditor = ({
     // Update the caret position in the input box when changed by a side effect
     useEffect(() => {
         const textbox: HTMLInputElement | HTMLTextAreaElement | undefined = textboxRef.current?.getInputBox();
-        if (textbox && textbox.selectionStart !== caretPosition) {
+        if (textbox && (textbox.selectionStart !== caretPosition || draft.message.length === 0)) {
```

Comment:
Do we want to add any test for this?

Reply:
Actually I'm changing this, sorry for the duplicate review but the current fix breaks lot of other things. Will update with a different set of changes.